inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp

```diff
@@ -140,13 +152,19 @@ std::vector<std::vector<std::uint8_t>> LayerTestsCommon::CalculateRefs() {
         std::copy(buffer, buffer + inputSize, referenceInput.data());
     }
 
-    auto ieOutPrc = outPrc;
+    const auto &actualOutputs = GetOutputs();
+    std::vector<InferenceEngine::Precision> ieOutPrc(actualOutputs.size(), outPrc);
     if (outPrc == InferenceEngine::Precision::UNSPECIFIED) {
-        const auto &actualOutputs = GetOutputs();
-        ieOutPrc = actualOutputs[0]->getTensorDesc().getPrecision();
+        for (size_t i = 0; i < ieOutPrc.size(); i++) {
+            ieOutPrc[i] = actualOutputs[i]->getTensorDesc().getPrecision();
+        }
+    }
+
+    std::vector<ngraph::element::Type_t> convertType(ieOutPrc.size());
+    for (size_t i = 0; i < ieOutPrc.size(); i++) {
```

Comment:
you can merge both cycles. Looks like duplication. And What is the reason of creation these vectors? I don't see using these variables.

Reply:
we need to create vectors of output types from IE network to convert NGraph output to their type
previous implementation convert all NGraph output to actualOutputs[0] type